File: tests/testdata/codegen/deferred-outputs-pp/nodejs/second.ts
import * as codeinfra from "@codeinfra/codeinfra";
import * as random from "@codeinfra/random";

interface SecondArgs {
    petName: codeinfra.Input<string>,
}

export class Second extends codeinfra.ComponentResource {
    public passwordLength: codeinfra.Output<number>;
    constructor(name: string, args: SecondArgs, opts?: codeinfra.ComponentResourceOptions) {
        super("components:index:Second", name, args, opts);
        const randomPet = new random.RandomPet(`${name}-randomPet`, {length: args.petName.length}, {
            parent: this,
        });

        const password = new random.RandomPassword(`${name}-password`, {
            length: 16,
            special: true,
            numeric: false,
        }, {
            parent: this,
        });

        this.passwordLength = password.length;
        this.registerOutputs({
            passwordLength: password.length,
        });
    }
}

File: tests/testdata/codegen/deferred-outputs-pp/python/second.py
import codeinfra
from codeinfra import Input
from typing import Optional, Dict, TypedDict, Any
import codeinfra_random as random

class SecondArgs(TypedDict, total=False):
    petName: Input[str]

class Second(codeinfra.ComponentResource):
    def __init__(self, name: str, args: SecondArgs, opts:Optional[codeinfra.ResourceOptions] = None):
        super().__init__("components:index:Second", name, args, opts)

        random_pet = random.RandomPet(f"{name}-randomPet", length=len(args["petName"]),
        opts = codeinfra.ResourceOptions(parent=self))

        password = random.RandomPassword(f"{name}-password",
            length=16,
            special=True,
            numeric=False,
            opts = codeinfra.ResourceOptions(parent=self))

        self.passwordLength = password.length
        self.register_outputs({
            'passwordLength': password.length
        })